hazelcast/src/main/java/com/hazelcast/internal/partition/operation/PartitionReplicaSyncRequestOffloadable.java

```diff
@@ -184,8 +183,7 @@ public void start() throws Exception {
             // while preparing replication operations
             if (!partitionStateManager.trySetMigratingFlag(partitionId)
                     && !partitionStateManager.isMigrating(partitionId)) {
```

Comment:
it seems to me, if `!partitionStateManager.trySetMigratingFlag(partitionId)` is true then `!partitionStateManager.isMigrating(partitionId)` has to be false, thus, this if condition is never entered. Am I missing something? 🤔

Reply:
You are right, assuming there was no concurrency. However since `PartitionReplicaSyncRequestOffloadable` is executed in generic operation threads and up to `ClusterProperty#PARTITION_MAX_PARALLEL_REPLICATIONS` can be executed concurrently, then it is possible that first method will return `false` and second one will also return `false`. 
That said, the execution of `isMigrating` flag management on the generic op thread complicates matters by introducing concurrency where there was none for no additional value. So I will pull the try-set and clear migrating flag logic into tasks to be executed on the partition thread and avoid concurrency issues.

Reply:
thanks. Since both of the accesses were to the same field and 1-2 liner methods, I thought JVM would inline and merge them, but I see that the field is volatile